File: Cafebook/CafebookModel/Model/ModelWeb/KhachHang/LichSuThueSachDto.cs
using System;
using System.Collections.Generic;

namespace CafebookModel.Model.ModelWeb.KhachHang
{
    public class LichSuThueSachDto
    {
        public int IdPhieuThueSach { get; set; }
        public DateTime NgayThue { get; set; }
        public DateTime? NgayHenTra { get; set; }
        public string TrangThai { get; set; } = string.Empty;
        public int SoLuongSach { get; set; }
        public decimal TongTienCoc { get; set; }

        public DateTime? NgayTra { get; set; }
        public decimal? TongPhiThue { get; set; }
        public decimal? TongTienPhat { get; set; }
        public decimal? TongTienCocHoan { get; set; }

        public bool LaSoTienTamTinh { get; set; }

        public List<ChiTietLichSuThueDto> ChiTietSachs { get; set; } = new();
    }

    public class ChiTietLichSuThueDto
    {
        public string TenSach { get; set; } = string.Empty;
        public int DoMoiKhiThue { get; set; }
        public string? GhiChuKhiThue { get; set; }
        public int? DoMoiKhiTra { get; set; }
        public string? GhiChuKhiTra { get; set; }
        public decimal TienPhatTre { get; set; }
        public decimal TienPhatHuHong { get; set; }
    }

    public class PagedLichSuThueSachResponseDto
    {
        public List<LichSuThueSachDto> Items { get; set; } = new List<LichSuThueSachDto>();
        public int TotalCount { get; set; }
        public int TotalPages { get; set; }
        public int CurrentPage { get; set; }
        public decimal PhatGiamDoMoi1Percent { get; set; }
    }
}

File: Cafebook/ADRCafebookApi/app/src/main/java/com/example/cafebook/models/LichSuThueSachDto.java
package com.example.cafebook.models;

import com.google.gson.annotations.SerializedName;
import java.io.Serializable;
import java.util.List;

public class LichSuThueSachDto {

    public static class Response {
        @SerializedName("items") public List<Item> items;
        @SerializedName("totalPages") public int totalPages;
        @SerializedName("currentPage") public int currentPage;
        @SerializedName("phatGiamDoMoi1Percent") public double phatGiamDoMoi1Percent;
    }

    public static class Item implements Serializable {
        @SerializedName("idPhieuThueSach") public int idPhieuThueSach;
        @SerializedName("ngayThue") public String ngayThue;
        @SerializedName("ngayHenTra") public String ngayHenTra;
        @SerializedName("trangThai") public String trangThai;
        @SerializedName("soLuongSach") public int soLuongSach;
        @SerializedName("tongTienCoc") public double tongTienCoc;
        @SerializedName("ngayTra") public String ngayTra;
        @SerializedName("tongPhiThue") public Double tongPhiThue;
        @SerializedName("tongTienPhat") public Double tongTienPhat;
        @SerializedName("tongTienCocHoan") public Double tongTienCocHoan;
        @SerializedName("laSoTienTamTinh") public boolean laSoTienTamTinh;
        @SerializedName("chiTietSachs") public List<Detail> chiTietSachs;
    }

    public static class Detail implements Serializable {
        @SerializedName("tenSach") public String tenSach;
        @SerializedName("doMoiKhiThue") public int doMoiKhiThue;
        @SerializedName("ghiChuKhiThue") public String ghiChuKhiThue;
        @SerializedName("doMoiKhiTra") public Integer doMoiKhiTra;
        @SerializedName("ghiChuKhiTra") public String ghiChuKhiTra;
        @SerializedName("tienPhatTre") public double tienPhatTre;
        @SerializedName("tienPhatHuHong") public double tienPhatHuHong;
    }
}
